src/test/java/uk/gov/hmcts/cmc/claimstore/rules/ClaimantRepaymentPlanRuleTest.java

```diff
@@ -0,0 +1,140 @@
+package uk.gov.hmcts.cmc.claimstore.rules;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.junit.MockitoJUnitRunner;
+import uk.gov.hmcts.cmc.claimstore.exceptions.ClaimantInvalidRepaymentPlanException;
+import uk.gov.hmcts.cmc.domain.models.Claim;
+import uk.gov.hmcts.cmc.domain.models.PaymentOption;
+import uk.gov.hmcts.cmc.domain.models.response.FullAdmissionResponse;
+import uk.gov.hmcts.cmc.domain.models.response.PartAdmissionResponse;
+import uk.gov.hmcts.cmc.domain.models.response.PaymentIntention;
+import uk.gov.hmcts.cmc.domain.models.sampledata.SampleClaim;
+import uk.gov.hmcts.cmc.domain.models.sampledata.SampleRepaymentPlan;
+import uk.gov.hmcts.cmc.domain.models.sampledata.SampleResponse;
+import uk.gov.hmcts.cmc.domain.models.sampledata.response.SamplePaymentIntention;
+
+import java.time.LocalDate;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ClaimantRepaymentPlanRuleTest {
+
+    private ClaimantRepaymentPlanRule claimantRepaymentPlanRule = new ClaimantRepaymentPlanRule();
+
+    private final PaymentIntention installmentPaymentIntentionBeforeOneMonth = SamplePaymentIntention.builder()
+        .paymentOption(PaymentOption.INSTALMENTS).repaymentPlan(
+            SampleRepaymentPlan.builder().withFirstPaymentDate(LocalDate.now().plusDays(3)).build())
+        .build();
+
+    private final PaymentIntention installmentPaymentIntentionAfterOneMonth = SamplePaymentIntention.builder()
+        .paymentOption(PaymentOption.INSTALMENTS).repaymentPlan(
+            SampleRepaymentPlan.builder().withFirstPaymentDate(LocalDate.now().plusMonths(2)).build())
+        .build();
+
+
```

Comment:
Tests should expect exception type as well as the exception message to make sure we are throwing the right exception with right message.